### DISCUSSION TITLE
Creating custom directive

### DISCUSSION THREAD

**Discussion by Bond-Addict** [epicmaxco/vuestic-ui#3576] [Q&A]
So I've been going down a deep rabbit hole on directives and I'd like to add one on ``<va-date-input>`` for example.

I have observed that putting the directive on the vuestic element causes the directive to not be hit. Based on my research its because its third party code which I don't have direct access to. I know that I can wrap it in a div and I will get the vnode for the div with children.

Is it possible to emit one of the emitOptions via this method? 

The code I've been working with is below:

```
import { Directive } from "vue";
export const vTest = {
  mounted(el, binding, vnode, prevVnode) {
    console.log(el, binding, vnode, prevVnode);
  },
  updated(el, binding, vnode, prevVnode) {
    var element = vnode.children[0];
  },
} as Directive;
```

I tried ``element.component.vnode.component.emit('update:is-open', false);`` which at a glance emit even knows the ``update:is-open`` event exists. 

So my question is, am I close, is this even possible?  I know there is a story in the backlog for closing the date picker on click, but I figured for the sake of expanding my knowledge I would see if I can create my own as I will undoubtedly create directives on other components as I continue to develop my application if it is indeed possible, so the knowledge would be very useful.

TIA!

**Comment by m0ksem** [comment]
We wrap components in proxy withTransportConfig... Maybe this is a problem. Need a research.
